pkg/edit/editor.go

```diff
@@ -102,6 +108,13 @@ func (ed *Editor) ReadCode() (string, error) {
 	return ed.app.ReadCode()
 }
 
+// RunAfterCommandHooks runs callbacks involving the interactive completion of a command line.
+func (ed *Editor) RunAfterCommandHooks(command string, duration float64, err error) {
```

Comment:
Instead of passing the command as a string, pass it as a `parse.Source`, a type that is already used to represent a piece of source code to execute.

Reply:
/facepalm and Homer Simpson "Doh!" Fixed. Which resulted in my deciding to also expose the "name" and "is-file" attributes as well as "code" (formerly "command").